lib/compress-stream.js

```diff
@@ -52,18 +60,25 @@ CompressStream.prototype._uncompressed = function (chunk) {
 
 CompressStream.prototype._transform = function (chunk, enc, callback) {
   var self = this
+  async function compressChunks() {
+    try {
+      for (let startFrom = 0; startFrom < chunk.length; startFrom += UNCOMPRESSED_CHUNK_SIZE) {
+        const endAt = startFrom + Math.min(chunk.length - startFrom, UNCOMPRESSED_CHUNK_SIZE);
+        const bytesChunk = chunk.slice(startFrom, endAt);
+        const compressed = snappy.compressSync(bytesChunk)
```

Comment:
Why switch to the sync version? If we want to do that we should do it in another PR as study in depth the implications

Reply:
@dapplion  the async version where I chunk bytes from outside (like call compressStream.write(chunk)) which uses the original flow, compared to this where chunking happen in the transform using this way is 2X better:
![image](https://user-images.githubusercontent.com/76567250/163566394-7e8f891a-7f5b-4655-8382-74873c8a97d9.png)
(orig-snappy is the original)

If I use async version inside with `await Promise((resolve)=>...` wrap to make sure the async returned data is written in a proper serial order, its 10-5% slower because of the overhead, if I don't care about the serial order in which chunks are written (which leads to incorrect stream), then async chunking version inside transform is 5% better.

I think for now this is our best bet :slightly_smiling_face:

Reply:
Okay sounds good to me. Can you commit a benchmark tho? To inform us of the cost of compressing an uncompressing objects of 10Kb, 100Kb, 1000Kb